src/backend/cuda/platform.cpp

```diff
@@ -332,10 +342,10 @@ DeviceManager::DeviceManager()
 
     sortDevices();
 
-    for(int i = 0; i < nDevices; i++) {
-        setActiveDevice(i, cuDevices[i].nativeId);
-        CUDA_CHECK(cudaStreamCreate(&streams[i]));
-    }
+    // Initialize all streams to 0.
+    // Streams will be created in setActiveDevice()
+    for(int i = 0; i < (int)MAX_DEVICES; i++)
+        streams[i] = (cudaStream_t)0;
```

Comment:
Creating cuda streams is taking about 800ms/device at initialization. So doing it for all available devices is making it slow on systems with multiple GPUs. Doing what you suggested does that same thing as before, simply in a different manner.

This change creates the stream when the device is switched, ie. set device and not when the code is executed.